commit 93769fef8d6122285b94b13fc8872847182f5440
Author: J. Bruce Fields <bfields@redhat.com>
Date:   Thu Feb 14 12:33:19 2019 -0500

    Revert "nfsd4: return default lease period"
    
    commit 3bf6b57ec2ec945e5a6edf5c202a754f1e852ecd upstream.
    
    This reverts commit d6ebf5088f09472c1136cd506bdc27034a6763f8.
    
    I forgot that the kernel's default lease period should never be
    decreased!
    
    After a kernel upgrade, the kernel has no way of knowing on its own what
    the previous lease time was.  Unless userspace tells it otherwise, it
    will assume the previous lease period was the same.
    
    So if we decrease this value in a kernel upgrade, we end up enforcing a
    grace period that's too short, and clients will fail to reclaim state in
    time.  Symptoms may include EIO and log messages like "NFS:
    nfs4_reclaim_open_state: Lock reclaim failed!"
    
    There was no real justification for the lease period decrease anyway.
    
    Reported-by: Donald Buczek <buczek@molgen.mpg.de>
    Fixes: d6ebf5088f09 "nfsd4: return default lease period"
    Cc: stable@vger.kernel.org
    Signed-off-by: J. Bruce Fields <bfields@redhat.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/fs/nfsd/nfsctl.c b/fs/nfsd/nfsctl.c
index 899174c7a8ae..39b835d7c445 100644
--- a/fs/nfsd/nfsctl.c
+++ b/fs/nfsd/nfsctl.c
@@ -1239,8 +1239,8 @@ static __net_init int nfsd_init_net(struct net *net)
 	retval = nfsd_idmap_init(net);
 	if (retval)
 		goto out_idmap_error;
-	nn->nfsd4_lease = 45;	/* default lease time */
-	nn->nfsd4_grace = 45;
+	nn->nfsd4_lease = 90;	/* default lease time */
+	nn->nfsd4_grace = 90;
 	nn->somebody_reclaimed = false;
 	nn->clverifier_counter = prandom_u32();
 	nn->clientid_counter = prandom_u32();
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1421_93769fef8d6122285b94b13fc8872847182f5440.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1421_93769fef8d6122285b94b13fc8872847182f5440.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
